smart_contract/executor/src/permission.rs

```diff
@@ -212,31 +212,40 @@ pub mod account {
 pub mod trigger {
     //! Module with pass conditions for trigger related tokens
     use super::*;
+    use crate::permission::domain::is_domain_owner;
 
     /// Check if `authority` is the owner of `trigger_id`.
     ///
     /// `authority` is owner of `trigger_id` if:
     /// - `trigger.action.authority` is `authority`
+    /// - `trigger.action.authority.domain_id` is owned by `authority`
     /// - `trigger.domain_id` is not none and domain is owned by `authority`
     ///
     /// # Errors
     /// - `FindTrigger` fails
     /// - `is_domain_owner` fails
     pub fn is_trigger_owner(trigger_id: &TriggerId, authority: &AccountId) -> Result<bool> {
-        let trigger = FindTriggerById::new(trigger_id.clone())
-            .execute()
-            .map(QueryOutputCursor::into_inner)?;
-        if trigger.action().authority() == authority {
+        let trigger = find_trigger(trigger_id)?;
+
+        if trigger.action().authority() == authority
+            || is_domain_owner(trigger.action().authority().domain_id(), authority)?
+            || match trigger_id.domain_id() {
+                Some(domain) => is_domain_owner(domain, authority)?,
+                None => false,
+            }
+        {
```

Comment:
If look unnecessary here because we essentially do here 

```rust
if true {
    return Ok(true);
} else {
    return Ok(false);
}
``` 

Instead we can use predicate directly:

```suggestion
        Ok(
            trigger.action().authority() == authority
            || is_domain_owner(trigger.action().authority().domain_id(), authority)?
            || match trigger_id.domain_id() {
                Some(domain) => is_domain_owner(domain, authority)?,
                None => false,
            }
        )
```

Reply:
Good advice, thanks!